src/dlls/mscoree/unixinterface.cpp

```diff
@@ -187,7 +167,130 @@ HRESULT ExecuteAssembly(
+//
+// Parameters:
+//  hostHandle              - Handle of the host
+//  domainId                - Id of the domain 
+//  entryPointAssemblyName  - Name of the assembly which holds the custom entry point
+//  entryPointTypeName      - Name of the type which holds the custom entry point
+//  entryPointMethodName    - Name of the method which is the custom entry point
+//  delegate                - Output parameter, the function stores a pointer to the delegate at the specified address
+//
+// Returns:
+//  HRESULT indicating status of the operation. S_OK if the assembly was successfully executed
+//
+extern "C"
+HRESULT CreateDelegate(
+            void* hostHandle,
+            DWORD domainId,
+            LPCSTR entryPointAssemblyName,
+            LPCSTR entryPointTypeName,
+            LPCSTR entryPointMethodName,
+            void** delegate)
+{
+    ICLRRuntimeHost2* host = reinterpret_cast<ICLRRuntimeHost2*>(hostHandle);
+
+    ConstWStringHolder entryPointAssemblyNameW = StringToUnicode(entryPointAssemblyName);
+    ConstWStringHolder entryPointTypeNameW = StringToUnicode(entryPointTypeName);
+    ConstWStringHolder entryPointMethodNameW = StringToUnicode(entryPointMethodName);
+
+    HRESULT hr = host->CreateDelegate(
+                            domainId,
+                            entryPointAssemblyNameW,
+                            entryPointTypeNameW,
+                            entryPointMethodNameW,
+                            (INT_PTR*)delegate);
+    
+    return hr;
+}
+
+//
+// Execute a managed assembly with given arguments
+//
+// Parameters:
+//  exePath                 - Absolute path of the executable that invoked the ExecuteAssembly
+//  coreClrPath             - Absolute path of the libcoreclr.so
+//  appDomainFriendlyName   - Friendly name of the app domain that will be created to execute the assembly
+//  propertyCount           - Number of properties (elements of the following two arguments)
+//  propertyKeys            - Keys of properties of the app domain
+//  propertyValues          - Values of properties of the app domain
+//  argc                    - Number of arguments passed to the executed assembly
+//  argv                    - Array of arguments passed to the executed assembly
+//  managedAssemblyPath     - Path of the managed assembly to execute (or NULL if using a custom entrypoint).
+//  enntyPointAssemblyName  - Name of the assembly which holds the custom entry point (or NULL to use managedAssemblyPath).
+//  entryPointTypeName      - Name of the type which holds the custom entry point (or NULL to use managedAssemblyPath).
+//  entryPointMethodName    - Name of the method which is the custom entry point (or NULL to use managedAssemblyPath).
+//  exitCode                - Exit code returned by the executed assembly
+//
+// Returns:
+//  HRESULT indicating status of the operation. S_OK if the assembly was successfully executed
+//
+extern "C"
+HRESULT ExecuteAssembly(
```

Comment:
I am wondering whether we should prefix all hosting methods with coreclr_ , like core_initialize, coreclr_create_delegate, etc. I believe it is the common naming convention for flat C APIs on Unix - it would make the hosting APIs look more native.